geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -37,85 +39,92 @@ trait IngestCommand[DS <: DataStore] extends DataStoreCommand[DS] with Interacti
+        val typeName = Option(params.featureName).getOrElse {
+          val existing = withDataStore(_.getTypeNames)
+          val fileName = Option(FilenameUtils.getBaseName(file.path))
+          val base = fileName.map(_.trim.replaceAll("[^A-Za-z0-9]+", "_")).filterNot(_.isEmpty).getOrElse("geomesa")
+          var name = base
+          var i = 0
+          while (existing.contains(name)) {
+            name = s"${base}_$i"
+            i += 1
           }
-          sft = SimpleFeatureTypes.renameSft(inferred._1, typeName)
-          inferredSftString = Some(SimpleFeatureTypes.toConfig(sft, includePrefix = false).root().render(renderOptions))
-          Command.user.info(s"Inferred schema: $typeName identified ${SimpleFeatureTypes.encodeType(sft)}")
+          name
         }
-        val converterString = inferred._2.root().render(renderOptions)
-        Command.user.info(s"Inferred converter:\n$converterString")
-        if (Prompt.confirm("Use inferred converter (y/n)? ")) {
-          if (Prompt.confirm("Persist this converter for future use (y/n)? ")) {
-            writeInferredConverter(sft.getTypeName, converterString, inferredSftString)
-          }
-          converter = inferred._2
-        } else {
-          Command.user.info("Please re-run with a valid converter")
-          return
+        sft = SimpleFeatureTypes.renameSft(inferredSft, typeName)
+        inferredSftString = Some(SimpleFeatureTypes.toConfig(sft, includePrefix = false).root().render(renderOptions))
+        Command.user.info(s"Inferred schema: $typeName identified ${SimpleFeatureTypes.encodeType(sft)}")
+      }
+      val converterString = inferredConverter.root().render(renderOptions)
+      Command.user.info(s"Inferred converter:\n$converterString")
+      if (Prompt.confirm("Use inferred converter (y/n)? ")) {
+        if (Prompt.confirm("Persist this converter for future use (y/n)? ")) {
+          writeInferredConverter(sft.getTypeName, converterString, inferredSftString)
         }
+        converter = inferredConverter
+      } else {
+        Command.user.info("Please re-run with a valid converter")
+        return
       }
+    }
 
-      if (sft == null) {
-        throw new ParameterException("SimpleFeatureType name and/or specification argument is required")
-      } else if (converter == null) {
-        throw new ParameterException("Converter config argument is required")
-      }
+    if (sft == null) {
+      throw new ParameterException("SimpleFeatureType name and/or specification argument is required")
+    } else if (converter == null) {
+      throw new ParameterException("Converter config argument is required")
+    }
 
-      createConverterIngest(sft, converter).run()
+    if (params.fmt == DataFormats.Shp) {
+      // shapefiles have to be ingested locally, as we need access to the related files
+      if (params.mode == RunModes.Distributed) {
+        Command.user.warn("Forcing run mode to local for shapefile ingestion")
```

Comment:
Nice.

What are the options around running a distributed ingest with shapefiles?  Could we split up the list of files among workers?

Reply:
I don't think we can realistically split up a shapefile among different workers. the other issue is that we'd have to zip up all the relevant files into a single file, as they have to be processed together.

Reply:
If all the files are on remote filesystems, I don't think we need to fuss with zipping up the files...

And yes, it'd be one or more shapefiles per task.  Splitting up a shapefile wouldn't make sense.

Reply:
the reason you'd have to zip it up is b/c you wouldn't get all the files into a single mapper if you don't... i guess we could create some kind of weird input format split that has the path to the rest of the files (which is pretty much what i did to hack shp into converters), but it doesn't really fit with the whole m/r paradigm